db/db_impl/db_impl_open.cc

```diff
@@ -1314,6 +1327,51 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
   return status;
 }
 
+void DBImpl::MoveCorruptedWalFiles(std::vector<uint64_t>& wal_numbers,
+                                   uint64_t corrupted_wal_number) {
+  size_t corrupt_index_start = wal_numbers.size() - 1;
+  size_t i = 0;
+
+  // Find the index of first corrupted wal.
+  while (i <= wal_numbers.size() - 1) {
+    if (wal_numbers[i] == corrupted_wal_number) {
+      corrupt_index_start = i;
+      break;
+    }
+    i++;
+  }
+
+  std::string archivalPath =
+      ArchivalDirectory(immutable_db_options_.GetWalDir());
+  Status create_status = env_->CreateDirIfMissing(archivalPath);
+
+  // Move all the WAL files from corrupted_wal_number to last WAL
+  // (max_wal_number) to avoid column family inconsistency error to archival
+  // directory. If its unable to create archive dir, it will delete the
+  // corrupted WAL files.
+  while (i <= wal_numbers.size() - 1) {
+    LogFileNumberSize log(wal_numbers[i]);
+    std::string fname =
+        LogFileName(immutable_db_options_.GetWalDir(), wal_numbers[i]);
+    if (create_status.ok()) {
+      wal_manager_.ArchiveWALFile(fname, wal_numbers[i]);
```

Comment:
@riversand963 I moved the files to an archive folder and if for some reason it fails to create folder, it deletes the files. Do you think it's too aggressive or it's fine? If for some reason folder is not created, it will generate inconsistency error when open after recovery.

Reply:
let's not delete the WALs in this case. We should only delete the WAL once DB figures out the WAL is not needed (< min_log_number_to_keep).